bokehjs/src/coffee/models/canvas/canvas.coffee

```diff
@@ -47,15 +47,23 @@ class CanvasView extends BokehView
     ctx = canvas_el[0].getContext('2d')
     return ctx
 
+  need_render: () ->
+    # Public method so the plot can determine if size/dpi changed
+    width = @mget('width')
+    height = @mget('height')
```

Comment:
In the new layout system, these may not be the right variables to check. I'm still ironing out the details of that. Is there a functional test so we can know if we got this right?